x-pack/plugins/apm/public/components/app/mobile/service_overview/metrics_map/get_http_requests_map_layer_list.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EMSFileSourceDescriptor,
+  LayerDescriptor as BaseLayerDescriptor,
+  VectorLayerDescriptor as BaseVectorLayerDescriptor,
+  AGG_TYPE,
+  LAYER_TYPE,
+  SOURCE_TYPES,
+} from '@kbn/maps-plugin/common';
+import { v4 as uuidv4 } from 'uuid';
+import type { MapsStartApi } from '@kbn/maps-plugin/public';
+import { i18n } from '@kbn/i18n';
+import {
+  CLIENT_GEO_COUNTRY_ISO_CODE,
+  CLIENT_GEO_REGION_ISO_CODE,
+  PROCESSOR_EVENT,
+  SPAN_SUBTYPE,
+  SPAN_TYPE,
+} from '../../../../../../common/es_fields/apm';
+import { APM_STATIC_DATA_VIEW_ID } from '../../../../../../common/data_view_constants';
+import { getLayerStyle, PalleteColors } from './get_map_layer_style';
+
+interface VectorLayerDescriptor extends BaseVectorLayerDescriptor {
+  sourceDescriptor: EMSFileSourceDescriptor;
+}
+
+const COUNTRY_NAME = 'name';
+const PER_COUNTRY_LAYER_ID = 'per_country';
+const PER_REGION_LAYER_ID = 'per_region';
+const HTTP_REQUEST_PER_COUNTRY = `__kbnjoin__count__${PER_COUNTRY_LAYER_ID}`;
+const HTTP_REQUESTS_PER_REGION = `__kbnjoin__count__${PER_REGION_LAYER_ID}`;
+
+const label = i18n.translate(
+  'xpack.apm.serviceOverview.embeddedMap.httpRequests.metric.label',
+  {
+    defaultMessage: 'HTTP requests',
+  }
+);
+
+export async function getHttpRequestsLayerList(maps?: MapsStartApi) {
+  const whereQuery = {
+    language: 'kuery',
+    query: `${PROCESSOR_EVENT}:span and ${SPAN_SUBTYPE}:http and ${SPAN_TYPE}:external`,
```

Comment:
You could use `ProcessorEvent.span`. Also, span type and subtype are used on the `es` query as well, they could be a constant / enum but up to you.

Reply:
agreed.

Reply:
fixed it here https://github.com/elastic/kibana/pull/150040/commits/e7dd9513cfb75b98918c226c351ae395c31f1de3